extensions/amp-analytics/0.1/variables.js

```diff
@@ -181,6 +181,46 @@ function matchMacro(string, matchPattern, opt_matchingGroupIndexStr) {
   return matches && matches[index] ? matches[index] : '';
 }
 
+/**
+ * This macro function allows arithmetic operations over other analytics variables.
+ *
+ * @param {string} leftOperand
+ * @param {string} rightOperand
+ * @param {string} operation
+ * @param {string} round If this flag is truthy the result will be rounded
+ * @return {number}
+ */
+function calcMacro(leftOperand, rightOperand, operation, round) {
+  const left = Number(leftOperand);
+  const right = Number(rightOperand);
+  let result = 0;
+  userAssert(!isNaN(left), 'CALC macro - left operand must be a number found');
+  userAssert(
+    !isNaN(right),
+    'CALC macro - right operand must be a number found'
+  );
+  switch (operation) {
+    case 'ADD':
+      result = left + right;
+      break;
+    case 'SUBTRACT':
+      result = left - right;
+      break;
+    case 'MULTIPLY':
+      result = left * right;
+      break;
+    case 'DIVIDE':
+      result = left / right;
```

Comment:
Can we add an assert here that right is not 0? I don't think we want infinity as a return value.

Reply:
I have added appropriate assertion